src/core/io/UART.cpp

```diff
@@ -11,6 +11,12 @@ UART::UART(Pin txPin, Pin rxPin, uint32_t baudrate) {
     this->baudrate = baudrate;
 }
 
+UART::UART(const UART& uart) {
```

Comment:
It is necessary for UARTTX to be initialized with the actual hardware UART without needing to specify the pins that aren't important to the wrapper.

Reply:
I know that `UARTTX` needs a constructor that accepts a `UART`, but why does `UART` need that?

Reply:
So because technically ThreadUART is an instance of UART, we use this copy constructor to fill out its fields based on the passed-in UART. We don't actually use these fields because we also store a pointer to the UART instance we got and use that to run all the non-overloaded methods, so we don't have to use halUART.